prost-derive/src/lib.rs

```diff
@@ -414,23 +412,20 @@ fn try_oneof(input: TokenStream) -> Result<TokenStream, Error> {
         }
     }
 
-    let mut tags = fields
+    if let Some((invalid_variant, _)) = fields.iter().find(|(_, field)| field.tags().len() > 1) {
+        bail!(
+            "invalid oneof variant {}::{}: oneof variants may only have a single tag",
+            ident,
+            invalid_variant
+        );
+    }
+    if let Some((duplicate_tag, _)) = fields
         .iter()
-        .flat_map(|&(ref variant_ident, ref field)| -> Result<u32, Error> {
-            if field.tags().len() > 1 {
-                bail!(
-                    "invalid oneof variant {}::{}: oneof variants may only have a single tag",
-                    ident,
-                    variant_ident
-                );
```

Comment:
the `Err` this returns is formerly always discarded via flattening the `Result`.

Reply:
i believe the intention here was originally to write this as `.map(..).collect::<Result<Vec<_>, _>>()?;`, but the nuance was lost